### ISSUE TITLE
Remove empty item?

### ISSUE THREAD

**Issue by mmahalwy** [final-form/react-final-form-arrays#28]
`Field` component has the `parse` prop that allows authors to change the value before storage. That being said, `FieldArray` does not have the same prop and makes it difficult to parse out empty items. This will need to be done after submitting which is not ideal.

Thoughts or ideas?

**Comment by erikras** [comment]
Can you describe your use case further? If an array field becomes empty, i.e. `''`, it should be removed from the array?

**Comment by mmahalwy** [comment]
@erikras that's correct. If, for whatever reason, the user adds another input but does not add any content, the array will be `['Something new', 'another thing', '']`. Ideally, we can parse out the empty string element.

See:
![image](https://user-images.githubusercontent.com/3749095/39489636-1ba1611a-4d3b-11e8-88dd-85dceb4fe776.png)

**Comment by likerRr** [comment]
@wespickett when we map through the array of fields we need to show empty results in order to give ability to a user to fill an input. 

@erikras +1 I'd like to have simpler and accurate solution rather then filter empty results in submit handler. It would be great to encapsulate array-related logic inside an array-fields component and have ability to parse the result before it reaches the submit handler. But at the same time it should be visible as empty on UI for the user.

**Comment by mhuxtable** [comment]
I'm currently building a mechanism to always ensure there's an empty final element in the array (to render an empty text area without the need for an "add" button). This request would make such a requirement trivial.

**Comment by tonymaibox** [comment]
I like this library. I'm running into the same issue. Any updates?
